### PR TITLE
Update to sbt 1.2.8 (adding --ignore-source-errors for javadoc)

### PR DESCRIPTION
**Pull request by raboof**
`--ignore-source-errors` makes 'sbt -Dakka.genjavadoc.enabled=true clean unidoc' ignore source errors.

The genjavadoc-generated code in `akka.stream.scaladsl.GraphDSL.Implicits.ReversePortsOps` in `scaladsl/GraphDSL.java` seems to be problematic, but since this is scaladsl anyway it seems reasonable to just exclude it from the javadoc.

Refs #26057, #26100

### CODE REVIEW

**Comment by raboof** [comment]
Next step is probably to run the build with a debugger attached, putting a breakpoint at `jdk.javadoc.internal.tool.Main#execute` and rummaging about how the error handling there works.

**Comment by helena** [comment]
@raboof want to assign #26100 to you since these are marked related now and this resolves that?

File: project/Doc.scala [hunk 1]
```diff
@@ -115,7 +115,7 @@ object UnidocRoot extends AutoPlugin {
       .getOrElse(sbtunidoc.ScalaUnidocPlugin)
 
   val akkaSettings = UnidocRoot.CliOptions.genjavadocEnabled.ifTrue(
-    Seq(javacOptions in (JavaUnidoc, unidoc) := Seq("-Xdoclint:none"))).getOrElse(Nil)
+    Seq(javacOptions in (JavaUnidoc, unidoc) := Seq("-Xdoclint:none", "--ignore-source-errors"))).getOrElse(Nil)
```
Comment (helena) [inline project/Doc.scala]: Nice, is `--ignore-source-errors` still undocumented?

**Review by helena — APPROVED** [review]
LGTM - the failing two tests are finicky, I've seen them fail and then pass. We need to look at that.

**Comment by helena** [comment]
PLS BUILD ALL

**Comment by helena** [comment]
PLS BUILD

**Comment by raboof** [comment]
@helena I'm happy to take on #26100 further (after next week), but I'm afraid this WIP PR is not quite ready to be merged yet: even with `--ignore-source-errors`, `sbt -Dakka.genjavadoc.enabled=true clean unidoc` does not yet produce a full javadoc HTML hierarchy on this branch yet (while it does on master)

File: project/Doc.scala (hunk 1 above)
Comment (raboof) [reply project/Doc.scala]: AFAICT yes

**Comment by raboof** [comment]
Updated: I think this should do it :crossed_fingers:

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by cunei** [comment]
Note: the flag `--ignore-source-errors` is not recognized by JDK 1.8.0_202, which means doc generation fails in that case.

**Comment by raboof** [comment]
@cunei you're right! This doesn't seem like a big deal since we require jdk9+ for releasing anyway, but might be good to mention in CONTRIBUTING.md. Added #26317

**Pull request merged**